### DISCUSSION TITLE
using oslcquery with custom DOORS NG attributes

### DISCUSSION THREAD

**Discussion by kpribadi** [IBM/ELM-Python-Client#53] [Q&A]
I created some custom attributes in the DOORS NG Project, for example "My Group" with associated URI for example: "https://domain.com/ns/rm#MYGROUP". How do I specify the "-s" syntax using the specified URI instead of the "My Group" attribute name? Likewise the URI could be autogenerated by DOORS NG, and I would like to use the autogenerated URI as well.

Thank you for any help..

**Comment by barny** [comment]
Hi - there currently isn't a way to specify things in -s or -q using RDF URIs, it's not something I've wanted to do so I'm interested why you want to do this?

**Comment by barny** [comment]
> URI could be autogenerated by DOORS NG

I've never heard of autogeneration of RDF URIs - they have to be manually specified in the type system for the component/project.

**Comment by barny** [comment]
oslcquery lets you refer to properties/values using their human-friendly name, which is usually a lot shorter than a URI and also easier to remember/type.

The main reason I can think of to allow use of RDF URIs by oslcquery would be when using OSLC query (which refers to a type) in a GC because then the same type can be found with that URI in all the contributions to the GC - this will require oslcquery to search the component for each contribution to find the type with the URI and execute different queries against each contribution. That's a lot of complicated work currently well out of scope :-(
